ldms/src/ldmsd/ldmsd.h

```diff
@@ -1019,6 +1032,73 @@ struct ldmsd_cfgobj_sampler {
 	pthread_t xthread; /* the exclusive thread */
 };
 
+/*
+ * Return the configuration object associated with the plugin
+ */
+struct ldmsd_cfgobj *ldmsd_plugin_cfg_get(ldmsd_plugin_handle_t handle);
+
+/*
+ * Record an opaque context pointer in the plugin configuration handle.
+ *
+ * In order to be multi-configuration capable, a plugin will need to
+ * maintain separate state (i.e. context) for each configuration. A
+ * plugin will allocate a context structure in the plugin constructor()
+ * and use this function to assign it. Subsequent calls to plugin functions
+ * will use the ldmsd_plugin_ctxt_get() function to return this pointer.
+ */
+void ldmsd_plugin_ctxt_set(ldmsd_plugin_handle_t handle, void *context);
+
+/*
+ * Retrieve the context pointer from the plugin configuration handle
+ * that was set by the ldmsd_plugin_ctxt_set() function.
+ *
+ * This is typically a pointer to the structure created in the
+ * plugin's constructor() function.
+ */
+void *ldmsd_plugin_ctxt_get(ldmsd_plugin_handle_t handle);
+
+/*
+ * Retreive a pointer to the plugin's configuration name. Plugin
+ * configuration names will be unique to a give configuration (in
+ * contrast to the plugin name).
+ *
+ * This function is usefull when informing the administrator of the
+ * configuration to which this message applies
+ */
+const char *ldmsd_plugin_cfg_name_get(ldmsd_plugin_handle_t handle);
```

Comment:
I would really like to see the new accessor function be in a separate header. I would like to start down the path of making it clearer to plugin authors which things are APIs that they should be using, and which things are ldmsd internals. Right now ldmsd.h is a giant blob that mixes both with no clear distinction.

Reply:
> I would really like to see the new accessor function be in a separate header. I would like to start down the path of making it clearer to plugin authors which things are APIs that they should be using, and which things are ldmsd internals. Right now ldmsd.h is a giant blob that mixes both with no clear distinction.

Ok, I'll do that. I think ldmsd.h has gotten out of hand, but I think we should incrementally move stuff out. However, I have seen the multiple header thing turn into a jungle where everything includes everything and effectively it's the same as one big header.